environments/qa/modules/oaeqaautomation/templates/redeploy.sh.erb

```diff
@@ -29,43 +42,62 @@ fi
 echo "Shutting down services..."
 <%= scripts_dir %>/shutdown.sh
 
-echo ""
+echo
 echo "Deleting data..."
 <%= scripts_dir %>/deletedata.sh
 
-echo ""
+# I'm having less hassles when the app & ux dirs are removed
+rm -rf "<%= app_root_dir %>" "<%= ux_root_dir %>"
+
+echo
 echo "Restoring services with puppet"
 puppet agent --onetime --no-daemonize --verbose
 
-echo ""
+echo
 echo "Performing production build (this will take a little while)"
 
 service hilary stop
 
 cd <%= ux_root_dir %>
-npm install -d --loglevel warn
 npm install -g grunt-cli --loglevel warn
 
-# Only cdnify QA0
-if [ "<%= web_domain %>" == "oae-qa0.oaeproject.org" ] ; then
-    CDN_URL="https://qa0-researchresearch-researchresearch.netdna-ssl.com" grunt
+# so...in theory a tarball release can be handled by the puppet run earlier
+# however if we like to use the master branch by default in qa0
+if [ "Z<%= ux_install_method %>" == "Zarchive" ]
+then
+    echo "Detected tarball install"
+
+    # reset config.ui path
+    # I suspect that the ux files jiggle around could be done to resolve this instead but this seems to work so far
+    sed -i "s/'path': '\/opt\/3akai-ux'/'path': '\/opt\/3akai-ux\/optimized'/" /opt/oae/config.js
+
 else
-    grunt
+    echo "Detected git master install"
+
+    npm install -d --loglevel warn
+
+    # Only cdnify QA0
+    if [ "<%= web_domain %>" == "oae-qa0.oaeproject.org" ] ; then
+        CDN_URL="https://qa0-researchresearch-researchresearch.netdna-ssl.com" grunt
+    else
+        grunt
+    fi
```

Comment:
The value and conditional application of this CDN URL should be managed by puppet. If we have a property such as `ui_cdn_url` that can be read by the deploy script, then we just do `CDN_URL="<%= ui_cdn_url %>"`.

In `qa0.json`, we simply set it to "https://qa0-researchresearch-researchresearch.netdna-ssl.com"